src/effects/lv2/LV2Effect.cpp

```diff
@@ -3197,208 +2793,173 @@ LV2Wrapper::~LV2Wrapper()
+      // Insert another pointer before the null
+      // Append a feature to the array, only for the plugin instantiation
+      // (features are also used elsewhere to instantiate the UI in the
+      // suil_* functions)
+      // It informs the plugin how to send work to another thread
+      features.insert(features.end() - 1, &tempFeature);
 
 #if defined(__WXMSW__)
    // Plugins may have dependencies that need to be loaded from the same path
    // as the main DLL, so add this plugin's path to the DLL search order.
-   const LilvNode *const libNode = lilv_plugin_get_library_uri(plugin);
-   const char *const libUri = lilv_node_as_uri(libNode);
-   char *libPath = lilv_file_uri_parse(libUri, NULL);
-   wxString path = wxPathOnly(libPath);
+   const auto libNode = lilv_plugin_get_library_uri(plugin);
+   const auto libUri = lilv_node_as_uri(libNode);
+   const auto libPath = lilv_file_uri_parse(libUri, nullptr);
+   const auto path = wxPathOnly(libPath);
    SetDllDirectory(path.c_str());
    lilv_free(libPath);
 #endif
 
-   mInstance = lilv_plugin_instantiate(plugin,
-                                       sampleRate,
-                                       reinterpret_cast<LV2_Feature **>(features.data()));
+   mInstance = lilv_plugin_instantiate(plugin, sampleRate, features.data());
 
 #if defined(__WXMSW__)
-   SetDllDirectory(NULL);
+   SetDllDirectory(nullptr);
 #endif
 
-   if (mEffect->mWantsWorkerInterface)
-   {
-      // Remove terminator
-      features.pop_back();
-
-      // Remove the worker interface feature
-      features.pop_back();
-
-      // Re-add terminator
-      mEffect->AddFeature(NULL, NULL);
-   }
-
    if (!mInstance)
-   {
-      return NULL;
-   }
+      return;
 
    mHandle = lilv_instance_get_handle(mInstance);
-
-   mOptionsInterface = (LV2_Options_Interface *)
-      lilv_instance_get_extension_data(mInstance, LV2_OPTIONS__interface);
-
-   mStateInterface = (LV2_State_Interface *)
-      lilv_instance_get_extension_data(mInstance, LV2_STATE__interface);
-
-   mWorkerInterface = (LV2_Worker_Interface *)
-      lilv_instance_get_extension_data(mInstance, LV2_WORKER__interface);
-
-   if (mEffect->mLatencyPort >= 0)
-   {
-      lilv_instance_connect_port(mInstance, mEffect->mLatencyPort, &mLatency);
-   }
-
+   mOptionsInterface = static_cast<const LV2_Options_Interface *>(
+      lilv_instance_get_extension_data(mInstance, LV2_OPTIONS__interface));
+   mStateInterface = static_cast<const LV2_State_Interface *>(
+      lilv_instance_get_extension_data(mInstance, LV2_STATE__interface));
+   mWorkerInterface = static_cast<const LV2_Worker_Interface *>(
+      lilv_instance_get_extension_data(mInstance, LV2_WORKER__interface));
+   if (mEffect.mLatencyPort >= 0)
+      lilv_instance_connect_port(mInstance, mEffect.mLatencyPort, &mLatency);
    if (mWorkerInterface)
       mThread = std::thread{
          std::mem_fn( &LV2Wrapper::ThreadFunction ), std::ref(*this)
       };
```

Comment:
Self-review of this reminds me of something else.

Commit `f95aa0cd2ffe4b754c31ccaceedbb11e9ead5f21` removed error checks on creation of a `wxThread` and left the construction of a `std::thread` above.

That's a loss of error checking, and it bothers me now.

But it's not just here, it is almost everywhere we create `std::thread`, that we don't check for standard exceptions from the constructor.  (Almost because I think @vsverchinsky may have been careful in the new code in lib-ipc).

I think we need a review of all constructions of `std::thread` and should write some graceful error handling for failure of each, and logging of the problem.

I count 11 constructions now, in 9 files.

Are you @crsib willing to take on that task?

Reply:
I agree that we should handle exceptions, but I don't see it as priority now. An exception means that:

* OS ran out of resources
* Audacity has started way to many threads
* Something bad happened while moving std::function or any of additional parameters

In the first two cases Audacity won't be able to run anyway, chances of latter outside the development are miniscule.

Let's create a task, but I don't think that the priority is higher than P3